rasa/utils/tensorflow/models.py

```diff
@@ -575,21 +588,30 @@ def _prepare_ffnn_layer(
         )
 
     def _prepare_dot_product_loss(
-        self, name: Text, scale_loss: bool, prefix: Text = "loss"
+        self, name: Text, scale_loss: bool, prefix: Text = "loss",
     ) -> None:
-        self._tf_layers[f"{prefix}.{name}"] = layers.DotProductLoss(
+        self._tf_layers[f"{prefix}.{name}"] = self.dot_product_loss_layer(
             self.config[NUM_NEG],
-            self.config[LOSS_TYPE],
-            self.config[MAX_POS_SIM],
-            self.config[MAX_NEG_SIM],
-            self.config[USE_MAX_NEG_SIM],
-            self.config[NEGATIVE_MARGIN_SCALE],
-            scale_loss,
+            loss_type=self.config[LOSS_TYPE],
+            mu_pos=self.config[MAX_POS_SIM],
+            mu_neg=self.config[MAX_NEG_SIM],
+            use_max_sim_neg=self.config[USE_MAX_NEG_SIM],
+            neg_lambda=self.config[NEGATIVE_MARGIN_SCALE],
+            scale_loss=scale_loss,
             similarity_type=self.config[SIMILARITY_TYPE],
             constrain_similarities=self.config[CONSTRAIN_SIMILARITIES],
             model_confidence=self.config[MODEL_CONFIDENCE],
         )
 
+    @property
+    def dot_product_loss_layer(self) -> tf.keras.layers.Layer:
+        """Returns the dot-product loss layer to use.
+
+        Returns:
+            The loss layer that is used by `_prepare_dot_product_loss`.
+        """
+        return layers.SingleLabelDotProductLoss
```

Comment:
All transformer based models inside Rasa use `SingleLabelDotProductLoss` right now except `IntentTED`. So, when constructing the model architecture graph, `_prepare_dot_product_loss` can directly call the `dot_product_loss_layer` property of the model instead of the model class overriding `_prepare_dot_product_loss` method.
cc @JEM-Mosig tagging you since you added this.